frontend/src/components/accountPage/UserListForm/UserListForm.tsx

```diff
@@ -0,0 +1,118 @@
+import { Input, Form, Button, Row } from 'antd';
+import { FC, useState } from 'react';
+import { User } from '../UserList/UserList';
+
+export interface IUserListFormProps {
+  onAddUser: (newUser: User) => void;
+  onCancel: () => void;
+}
+const regexEmail = /^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+(.[a-zA-Z0-9-])+$/;
+
+const UserListForm: FC<IUserListFormProps> = props => {
+  const { onAddUser, onCancel } = props;
+  const [validForm, setValidForm] = useState<boolean>(false);
+  const [form] = Form.useForm();
+
+  const cancelForm = () => {
+    form.resetFields();
+    onCancel();
+  };
+
+  // Used to disable Save button
+  const handleChange = async (_: any, user: User) => {
+    try {
+      await form.validateFields(['userid', 'name', 'surname', 'email']);
+      setValidForm(true);
+    } catch (_) {
+      setValidForm(false);
+    }
+  };
+  const submitForm = (user: User) => {
+    onAddUser(user);
+    form.resetFields();
+  };
+
+  return (
+    <Form
+      form={form}
+      labelCol={{ span: 4 }}
+      wrapperCol={{ span: 24 }}
+      onFinish={submitForm}
+      onValuesChange={handleChange}
+    >
+      <Form.Item
+        name="userid"
+        label="User ID"
+        validateTrigger="onBlur"
+        rules={[
+          {
+            required: true,
+            message: 'Field required',
+          },
+        ]}
```

Comment:
```suggestion
        rules={[{
          required: true,
          message: 'Field required',
        }]}
```
I'd keep it compact wherever possible, JSX is already super intendous on its own (provided eslint doesn't complain)...
(same thing for the other fields)

Reply:
I tried to keep the syntax more compact but eslint complained..

Reply:
Since this is used more than once though you could make something like `const requiredFieldRuleSet = [{...}];` before the jsx in order to reuse it